EMF metamodel changes: localGenerated not 'unsettable' and extensionID not 'required'

cfsnook opened the pull request:
EMF metamodel changes:
1) localGenerated is no longer 'unsettable' (this simplifies things, makes it consistent with other attributes s.a. generated and gets rid of a generation error where there was no default - false)
2) extensionID is no longer required - this is because we are now using AbstractExtension for things that don't need IDs and it was a nuisance to have to write an ID in XEventB.
3) remove deprecated getURI method. has been deprecated for some time as it is just as easy to use the EcoreUtil's getURI
4) when getting the internal reference - check for 'isSet' rather than null

Dana: Please check XEventB works better - i.e. you no longer need the string 'Includes' to set extensionID.
Son: general review

Commit:
minor formatting - no code change

Commit:
make localGenerated Attribute non-unsettable and provide default="false"

(re-generation has also changed default for internalId from null to "")

Commit:
remove deprecated getURI method

Commit:
Make attribute AbstractExtension:extensionID not-required (lower card 0)

Commit:
update notes files

Commit:
remove some unwanted old files

dd4g12 commented:
I tried the include without the extension id, it didn't complain but I had problems with running xeventb, its probably not related to this update because even when I changed things back to original event-b emf the problem was still there. Currently trying to figure out what is the problem with xevent-b.

Commit:
when getting internalID, check for isSet rather than null

At last re-generation the default for internalId changed from null to ""
This means that checking for null no longer works. In any case it is
more correct to call isSetInternalId as it is an unsettable attribute

cfsnook commented:
I discovered a bug in the latest changes while running the state machine unit tests. it is due to the re-generation changing the internal ID default from null to "".  Therefore it is important to test 'isSetInternalId' rather than check for null - this is better anyway.

Commit:
Update feature to use rodinp licence plugin etc.

Commit:
update sdk build and properties

cfsnook commented:
I have now updated the feature licensing etc. and updated the sdk feature build

Commit:
improve feature description and release history

Commit:
improve feature description and release history

Commit:
Merge https://github.com/cfsnook/EMF_EventB
